docs/pages/principles/testing.md

```diff
@@ -0,0 +1,94 @@
+## Unit Tests
```

Comment:
As mentioned above, I'd make this three sections. One short one about Integration testing. I think we could discuss ideas of cross-package integration testing there, like how to test that downstream packages still work, etc. Half of the below would go under "Unit tests", and the other half under the new "smoke" tests.